### PR TITLE
Add back `Update-List` command

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

Re-enable `Update-List` cmdlet bringing parity with WinPS5.1.  Added tests.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/5009
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj [hunk 1]
```diff
@@ -33,12 +33,10 @@
     <Compile Remove="commands\utility\ShowCommand\ShowCommandParameterSetInfo.cs" />
```
Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-static-analysis`
Verify Markdown Links.Verify links in /home/vsts/work/1/s/docs/learning-powershell/powershell-beginners-guide.md.https://examples.oreilly.com/9780596528492/ should work
```powershell
retry of URL failed with error: Connection timed out
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 117
117:                                     throw "retry of URL failed with error: $($_.Exception.Message)"

```

Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj]: @SteveL-MSFT, your last commit had 2 failures in `PowerShell-CI-macos`
Verify approved aliases list.All approved Cmdlets present (no new Cmdlets added, no Cmdlets removed)
```powershell
Expected $null or empty, but got Update-List.
at <ScriptBlock>, /Users/runner/runners/2.159.2/work/1/s/test/powershell/engine/Basic/DefaultCommands.Tests.ps1: line 584
584:         $result | Should -BeNullOrEmpty
```
Verify approved aliases list.All present Cmdlets should have the correct ConfirmImpact
```powershell
Expected $null or empty, but got @(@{Name=Write-Warning; ConfirmImpact=None; SideIndicator=<=}).
at <ScriptBlock>, /Users/runner/runners/2.159.2/work/1/s/test/powershell/engine/Basic/DefaultCommands.Tests.ps1: line 606
606:         $result | Should -BeNullOrEmpty
```

Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-windows`
Verify approved aliases list.All present Cmdlets should have the correct ConfirmImpact
```powershell
Expected $null or empty, but got @(@{Name=Update-List; ConfirmImpact=Medium; SideIndicator==>}, @{Name=Update-List; ConfirmImpact=None; SideIndicator=<=}).
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Basic\DefaultCommands.Tests.ps1: line 606
606:         $result | Should -BeNullOrEmpty
```

Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj]: @SteveL-MSFT, your last commit had 2 failures in `PowerShell-CI-linux`
Verify approved aliases list.All approved Cmdlets present (no new Cmdlets added, no Cmdlets removed)
```powershell
Expected $null or empty, but got Update-List.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/engine/Basic/DefaultCommands.Tests.ps1: line 584
584:         $result | Should -BeNullOrEmpty
```
Verify approved aliases list.All present Cmdlets should have the correct ConfirmImpact
```powershell
Expected $null or empty, but got @(@{Name=Write-Warning; ConfirmImpact=None; SideIndicator=<=}).
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/engine/Basic/DefaultCommands.Tests.ps1: line 606
606:         $result | Should -BeNullOrEmpty
```

**Comment by SteveL-MSFT** [comment]
@sdwheeler do you have an updated fwlink for this cmdlet?

**Comment by sdwheeler** [comment]
@SteveL-MSFT `https://go.microsoft.com/fwlink/?linkid=2109383`

File: src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj (hunk 1 above)
Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-macos`
Online help tests for PowerShell Cmdlets.Validate 'get-help Update-List -Online'
```powershell
Expected strings to be the same, but they were different.
Expected length: 46
Actual length:   47
Strings differ at index 40.
Expected: '...nkID=11344...'
But was:  '...nkID=21093...'
at <ScriptBlock>, /Users/runner/runners/2.159.2/work/1/s/test/powershell/engine/Help/HelpSystem.OnlineHelp.Tests.ps1: line 42
42:                 $actualURI | Should -Be $cmdlet.HelpURI
```

Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-windows`
Online help tests for PowerShell Cmdlets.Validate 'get-help Update-List -Online'
```powershell
Expected strings to be the same, but they were different.
Expected length: 46
Actual length:   47
Strings differ at index 40.
Expected: '...nkID=11344...'
But was:  '...nkID=21093...'
at <ScriptBlock>, D:\a\1\s\test\powershell\engine\Help\HelpSystem.OnlineHelp.Tests.ps1: line 42
42:                 $actualURI | Should -Be $cmdlet.HelpURI
```

Comment (PoshChan) [inline src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj]: @SteveL-MSFT, your last commit had 1 failures in `PowerShell-CI-linux`
Online help tests for PowerShell Cmdlets.Validate 'get-help Update-List -Online'
```powershell
Expected strings to be the same, but they were different.
Expected length: 46
Actual length:   47
Strings differ at index 40.
Expected: '...nkID=11344...'
But was:  '...nkID=21093...'
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/engine/Help/HelpSystem.OnlineHelp.Tests.ps1: line 42
42:                 $actualURI | Should -Be $cmdlet.HelpURI
```

**Review by anmenaga — APPROVED** [review]
Any info on why this cmdlet was removed originally?

**Comment by SteveL-MSFT** [comment]
I don't know why it was originally removed since it should have been compatible since PSCore6.  I don't find this cmdlet particularly useful (since you can just call the .NET API in PSScript), but a search on GitHub shows some scripts use it, so we should add it back for parity with WinPS.

**Pull request merged**